modules/govuk/manifests/apps/email_alert_service.pp

```diff
@@ -29,37 +29,47 @@
 #   Bearer token for communication with the email-alert-api
 #
 class govuk::apps::email_alert_service(
+  $enabled = false,
   $rabbitmq_hosts = ['localhost'],
   $rabbitmq_user = 'email_alert_service',
   $rabbitmq_password = 'email_alert_service',
   $sentry_dsn = undef,
   $redis_host = undef,
   $email_alert_api_bearer_token = undef,
 ) {
-  govuk::app { 'email-alert-service':
-    app_type           => 'bare',
-    enable_nginx_vhost => false,
-    sentry_dsn         => $sentry_dsn,
-    command            => './bin/email_alert_service',
-  }
+  if $enabled {
+    govuk::app { 'email-alert-service':
+      app_type           => 'bare',
+      enable_nginx_vhost => false,
+      sentry_dsn         => $sentry_dsn,
+      command            => './bin/email_alert_service',
+    }
 
-  Govuk::App::Envvar {
-    app => 'email-alert-service',
-  }
+    Govuk::App::Envvar {
+      app => 'email-alert-service',
+    }
 
-  govuk::app::envvar::rabbitmq { 'email-alert-service':
-    hosts    => $rabbitmq_hosts,
-    user     => $rabbitmq_user,
-    password => $rabbitmq_password,
-  }
+    govuk::app::envvar::rabbitmq { 'email-alert-service':
+      hosts    => $rabbitmq_hosts,
+      user     => $rabbitmq_user,
+      password => $rabbitmq_password,
+    }
 
-  govuk::app::envvar::redis { 'email-alert-service':
-    host => $redis_host,
-  }
+    govuk::app::envvar::redis { 'email-alert-service':
+      host => $redis_host,
+    }
 
-  govuk::app::envvar {
-    "${title}-EMAIL_ALERT_API_BEARER_TOKEN":
-        varname => 'EMAIL_ALERT_API_BEARER_TOKEN',
-        value   => $email_alert_api_bearer_token;
+    govuk::app::envvar {
+      "${title}-EMAIL_ALERT_API_BEARER_TOKEN":
+          varname => 'EMAIL_ALERT_API_BEARER_TOKEN',
+          value   => $email_alert_api_bearer_token;
+    }
+  } else {
+    govuk::app { 'email-alert-service':
+      ensure             => 'absent',
+      app_type           => 'bare',
+      enable_nginx_vhost => false,
+      command            => './bin/email_alert_service',
+    }
```

Comment:
This looks nasty, but I think that's just my general opinion on the imperative approach employed by Puppet, and we can't fix using puppet in this PR....

Anyway, an alternative would be to do something like this [1], where the enabled parameter feeds in to the ensure parameter for things in the class. That would also take care of the environment variables (which this big if currently doesn't), and be less nested. What do you think?

1: https://github.com/alphagov/govuk-puppet/blob/42147acc2610280b0ab34a1240fde87421cf1f05/modules/govuk_cdnlogs/manifests/transition_logs.pp#L37-L40

Reply:
That does look nicer, I'll see what I can do with that. 

The environment variables do get removed. I think puppet must fully control that dir.

Reply:
> This looks nasty, but I think that's just my general opinion on the imperative approach employed by Puppet, and we can't fix using puppet in this PR....

I shouldn't have said "nasty" here, that was inappropriate, so sorry about that.

> That does look nicer, I'll see what I can do with that.
> 
> The environment variables do get removed. I think puppet must fully control that dir.

Great, I'll take another look now.